tools/z64compress_wrapper.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: © 2024 ZeldaRET
+# SPDX-License-Identifier: CC0-1.0
+
+from __future__ import annotations
+
+import argparse
+import itertools
+import subprocess
+from pathlib import Path
+
+from elftools.elf.elffile import ELFFile
+from elftools.elf.sections import SymbolTableSection
+
+
+def ranges(i: list[int]):
+    for _, b in itertools.groupby(enumerate(i), lambda pair: pair[1] - pair[0]):
+        b = list(b)
+        yield b[0][1], b[-1][1]
+
+
+def get_compressed_segment_ranges(spec_path: Path) -> list[tuple[int, int]]:
+    spec = spec_path.read_text()
+    compress_segments = []
+
+    shift = 0
+    for i, segment in enumerate(spec.split("beginseg\n")[1:], 0):
+        directives = segment.split("endseg")[0].split("\n")
+
+        for directive in directives:
+            directive = directive.strip()
+            if directive.startswith("flags") and "NOLOAD" in directive:
+                shift += 1
+            elif directive.startswith("compress"):
+                compress_segments.append(i - shift)
+
+    return list(ranges(compress_segments))
```

Comment:
idk if this is the case in MM but in oot-gc-eu-mq-dbg the dmadata and the spec aren't ordered exactly the same
(`include/tables/dmadata_table_mqdbg.h` is used unless `NON_MATCHING`)

Tharo https://discord.com/channels/688807550715560050/688851317593997489/1181707645934121020
> map_i_static and map_48x85_static are reordered, and syotes is the first scene in the matching layout while in the spec it's several scenes down

so this code relying on spec order to tell which dma entries are compressed only works "by chance" kinda :sweat_smile:

Reply:
arrrrggg

I guess a "proper" fix for this issue would be to gather all the relocations from the dmatable from the built elf to know what are the proper indices.

Would you prefer if we tried to fix this now or in the future?

Reply:
since this is only an "issue" for matching builds but that it works anyway, I think this is fine to not fix at all. but it should be documented

or I may fix it if we yeet z64compress like I would like to